cpp/src/arrow/dataset/dataset_test.cc

```diff
@@ -471,8 +471,8 @@ class TestSchemaUnification : public TestDataset {
     };
 
     schema_ = SchemaFromNames({"phy_1", "phy_2", "phy_3", "phy_4", "part_ds", "part_df"});
-    ASSERT_OK_AND_ASSIGN(auto ds1, get_source("/dataset/alpha"));
-    ASSERT_OK_AND_ASSIGN(auto ds2, get_source("/dataset/beta"));
+    ASSERT_OK_AND_ASSIGN(auto ds1, get_source("/dataset/alpha", {ds1_df1, ds1_df2}));
```

Comment:
Was this changed due to ordering?

Reply:
I just changed it to use construction from an explicit file list rather than using a selector, so it's hitting the new code